spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryClient.java

```diff
@@ -73,10 +73,14 @@ public String description() {
 	}
 
 	private void addInstancesToList(List<ServiceInstance> instances, String serviceId, QueryParams queryParams) {
-
-		HealthServicesRequest request = HealthServicesRequest.newBuilder().setTag(this.properties.getDefaultQueryTag())
+		HealthServicesRequest.Builder requestBuilder = HealthServicesRequest.newBuilder()
 				.setPassing(this.properties.isQueryPassing()).setQueryParams(queryParams)
-				.setToken(this.properties.getAclToken()).build();
+				.setToken(this.properties.getAclToken());
+		String queryTag = this.properties.getQueryTagForService(serviceId);
+		if (queryTag != null) {
```

Comment:
Before we were setting this unconditionally (even when null) so what was getting passed into Consul health client was `tags=[null]` but they must handle that properly. I went ahead and only set when its specified.